.github/workflows/upload-release-to-plugin-repo.yml

```diff
@@ -5,10 +5,63 @@ on:
         types: [published]
 
 jobs:
+    compute_should_update_trunk:
+        runs-on: ubuntu-latest
+        outputs:
+            should_update_trunk: ${{ steps.compute-should-update-trunk.outputs.should_update_trunk }}
+        steps:
+            - name: Decide if trunk or tag
+              id: compute-should-update-trunk
+              uses: actions/github-script@v6
+              with:
+                  script: |
+                      const https = require('https');
+                      const repo = process.env.GITHUB_REPOSITORY;
+                      const githubRef = context.ref;
+
+                      function fetchJson(url) {
+                        return new Promise((resolve, reject) => {
+                          https.get(url, { headers: { Accept: 'application/json', 'User-Agent': 'Github Action Script' } }, res => {
+                            let data = '';
+                            res.on('data', chunk => data += chunk);
+                            res.on('end', () => resolve(JSON.parse(data)));
+                          }).on('error', reject);
+                        });
+                      }
+
+                      /**
+                       * Compares two version strings.
+                       *
+                       * @param {string} v1 - The first version string to compare, which may have an optional leading "v".
+                       * @param {string} v2 - The second version string to compare, which may have an optional leading "v".
+                       * @returns {number} - Returns 1 if v1 is greater than v2, -1 if v1 is less than v2, and 0 if v1 and v2 are equal.
+                       */
+                      function compareVersions(v1, v2) {
+                        const p1 = v1.replace(/^v/, '').split('.').map(Number), p2 = v2.replace(/^v/, '').split('.').map(Number);
+                        return p1.reduce((acc, part, i) => acc === 0 ? (part > p2[i] ? 1 : (part < p2[i] ? -1 : 0)) : acc, 0);
+                      }
+
+                      (async () => {
+                        const [githubData, wpData] = await Promise.all([
+                          fetchJson(`https://api.github.com/repos/${repo}/releases/latest`),
+                          fetchJson('https://api.wordpress.org/plugins/info/1.2/?action=plugin_information&request[slug]=gutenberg'),
+                        ]);
+
+                        const latestVersionInGithub = githubData.tag_name
```

Comment:
I think that a minor improvement (for visibility) would be to fetch these two versions in different jobs that this job would depend on, to give more visibility during the workflow runs (if debug logging is enabled) 🤔

Reply:
Actually, I found a logic mistake here. I'm still relying on the `latest` release on GH, and I should just rely on the published ref to compare against the version on core. Fixing it now, will submit a change that will also simplify the job not to use JS and will split it into a few more jobs to make debugging easier.